util/types/convert.go

```diff
@@ -127,6 +136,10 @@ func Convert(val interface{}, target *FieldType) (v interface{}, err error) { //
 		if err != nil {
 			return InvConv(val, tp)
 		}
+		// TODO: We should first convert to uint64 then check unsigned flag.
+		if mysql.HasUnsignedFlag(target.Flag) {
```

Comment:
Negative number will be converted to a big uint.

mysql> select cast(-5 as unsigned);
+----------------------+
| cast(-5 as unsigned) |
+----------------------+
| 18446744073709551611 |
+----------------------+
1 row in set (0.00 sec)

mysql>